packages/mds-logger/index.ts

```diff
@@ -19,27 +19,41 @@ import httpContext from 'express-http-context'
 const logger: Pick<Console, 'info' | 'warn' | 'error'> = console
 type LogLevel = keyof typeof logger
 
-const redact = (args: unknown[]): string[] =>
-  args.map(arg =>
-    JSON.stringify(arg instanceof Error ? arg.toString() : arg, (k, v) =>
-      ['lat', 'lng'].includes(k) ? '[REDACTED]' : v
-    )
+type LogArgs = [message: string, data?: Record<string, unknown> | Error]
+
+const redact = (arg: unknown): any => {
+  if (arg === undefined) return {}
+  const res = JSON.stringify(arg instanceof Error ? { error: arg.toString() } : arg, (k, v) =>
+    ['lat', 'lng'].includes(k) ? '[REDACTED]' : v
   )
 
-const log = (level: LogLevel, ...args: unknown[]): string[] => {
-  const redacted = process.env.QUIET === 'true' ? [] : redact(args)
-  if (redacted.length) {
-    const timestamp = Date.now()
-    const ISOTimestamp = new Date(timestamp).toISOString()
-    const requestId = httpContext.get('x-request-id')
+  return JSON.parse(res)
+}
 
-    logger[level](level.toUpperCase(), ISOTimestamp, timestamp, ...(requestId ? [requestId] : []), ...redacted)
+const log = (level: LogLevel, ...[message, data]: LogArgs): { log_message?: string; log_data?: any } => {
+  if (process.env.QUIET === 'true') {
+    return {}
   }
-  return redacted
+
+  const { log_message, log_data } = { log_message: redact(message), log_data: redact(data) }
```

Comment:
Why `log_message, log_data` instead of just `message, data` ... the `log_` prefix seems unnecessary.

Reply:
Was in an attempt to keep these keys from conflicting with other unpacked metadata that's used for indexing (example, fluent-bit adds some of its own annotations too). Because of how brittle ES can be around conflicting keys in a message, I'm playing it suuuuuper safe. There could be a better prefix, though, or we could wrap all of these things inside of a parent object with a key`mds_log` or something.